components/camel-hazelcast/src/main/java/org/apache/camel/component/hazelcast/topic/HazelcastTopicProducer.java

```diff
@@ -41,19 +42,18 @@ public HazelcastTopicProducer(HazelcastInstance hazelcastInstance, HazelcastDefa
     }
 
     public void process(Exchange exchange) throws Exception {
-        final int operation = lookupOperationNumber(exchange);
+        final HazelcastOperation operation = lookupOperation(exchange);
 
-        switch (operation) {
-        case -1:
-            // default operation to publish
-        case HazelcastConstants.PUBLISH_OPERATION:
+        switch ((operation != null) ? operation : HazelcastOperation.PUBLISH) {
```

Comment:
It would be nice if the default operation is set on the endpoint so it can be configured and lookupOperation would always return a non null value